tensorboard/plugins/graph/polymer3/tf_graph/tf-graph-scene.ts

```diff
@@ -0,0 +1,670 @@
+  // the DOM). We do not fit the hierarchy again (unless the user clicks the reset button). For
+  // instance, if the user enters a certain view in the graph, switches to another dashboard, and
+  // returns to the graph dashboard, the user expects the previous view. These properties enable
+  // that behavior.
+
+  /** Whether the scene has fit the current render hierarchy (to the viewport) at least once. */
+  @property({type: Boolean})
+  _hasRenderHierarchyBeenFitOnce: boolean;
+
+  /** Whether this scene element is currently attached to a parent element. */
+  @property({type: Boolean})
+  _isAttached: boolean;
+
+  /** This property is a d3_zoom object. */
+  @property({type: Object})
+  _zoom: object;
+  @property({
+    type: String,
+    observer: '_highlightedNodeChanged',
+  })
+  highlightedNode: string;
+  @property({
+    type: String,
+    observer: '_selectedNodeChanged',
+  })
+  selectedNode: string;
+
+  // An optional callback that implements the tf.graph.edge.EdgeSelectionCallback signature. If
+  // provided, edges are selectable, and this callback is run when an edge is selected.
+  @property({type: Object})
+  handleEdgeSelected: object;
+
+  /** Keeps track of if the graph has been zoomed/panned since loading */
+  @property({
+    type: Boolean,
+    observer: '_onZoomChanged',
+  })
+  _zoomed: boolean = false;
+
+  /**
+   * Keeps track of the starting coordinates of a graph zoom/pan.
+   *
+   * @private {{x: number, y: number}?}
+   */
+  @property({
+    type: Object,
+  })
+  _zoomStartCoords: object = null;
+
+  /**
+   * Keeps track of the current coordinates of a graph zoom/pan
+   *
+   * @private {{x: number, y: number}?}
+   */
+  @property({
+    type: Object,
+  })
+  _zoomTransform: object = null;
+
+  /** Maximum distance of a zoom event for it to be interpreted as a click */
+  @property({
+    type: Number,
+  })
+  _maxZoomDistanceForClick: number = 20;
+
+  /**
+   * Scale mapping from template name to a number between 0 and N-1
+   * where N is the number of different template names. Used by
+   * tf_graph_scene_node when computing node color by structure.
+   * This property is a d3.scale.ordinal object.
+   */
+  @property({type: Object})
+  templateIndex: Function;
+
+  /**
+   * A minimap object to notify for zoom events.
+   * This property is a tf.scene.Minimap object.
+   */
+  @property({type: Object})
+  minimap: object;
+
+  /*
+   * Dictionary for easily stylizing nodes when state changes.
+   * _nodeGroupIndex[nodeName] = d3_selection of the nodeGroup
+   */
+  @property({
+    type: Object,
+  })
+  _nodeGroupIndex = {};
+
+  /*
+   * Dictionary for easily stylizing annotation nodes when state changes.
+   * _annotationGroupIndex[nodeName][hostNodeName] =
+   *   d3_selection of the annotationGroup
+   */
+  @property({
+    type: Object,
+  })
+  _annotationGroupIndex: object = function() {
+    return {};
+  };
```

Comment:
few function that returns props value here.

Reply:
Thanks, done.